### ISSUE TITLE
[Bug]: Recently-added bookmarks are hard to use with a screen reader

### ISSUE THREAD

**Issue by DiegoARL38** [mozilla-mobile/fenix#21489]
### Steps to reproduce

1. Open Firefox with TalkBack enabled.
2. Navigate to any URL and add it to bookmarks.
3. Go to the browser's homepage and explore it via touch, or with TalkBack's navigation gestures.

### Expected behaviour

* Recently-added bookmark entries should be read as one item each, with both it's name and shortened URL, similar to how the Jump back to section entries are read.
* If exploring via touch, the area of each entry should be as big as the entry itself, including it's icon.

### Actual behaviour

* When exploring each entry, one is able to reach both the bookmark's name and it's shortened URL as distinct, separate items.
* It's quite hard to land on each entry if using touch exploration, since the area that the text occupies is very small and doesn't reflect the whole entry, ignoring the icon.

### Device name

Xiaomi Redmi Note 8

### Android version

MIUI 12.0.4.0, Android 10

### Firefox release type

Firefox Nightly

### Firefox version

94.0a1 (Build #2015835723) from 2021-09-23T17:08:56.236491

### Additional information

TalkBack version 9.1

**Comment by SoftVision-LorandJanos** [comment]
Hi @DiegoARL38! Thank you for your report!
I can confirm this issue is reproducible with the following detail: If talkback is activated for the recently saved bookmarks section, it just presents the title, it only reads the bookmark titles when focusing on them.

Firefox versions tested: latest Nightly (94.0a1 - 2021-09-24T05:10:09.860159)

Devices used:

- Samsung Galaxy Tab S3 (Android 9).
- Huawei P30 (Android 10).
- Xiaomi Mi11 Lite (Android 11).

**Comment by DiegoARL38** [comment]
The issue appears to be fixed since Nightly 94.0a1 (Build #2015835915)
   from 2021-09-24T17:07:58.548822.

Leaving opened for verification. If not needed, I'll close it.

**Comment by SoftVision-LorandJanos** [comment]
Works as intended on the latest Nightly 98.0a1 (2022-01-12).
Device used: Google Pixel 4 (Android 12).
Closing the ticket as fixed.
